chainerx_cc/chainerx/routines/reduction.cc

```diff
@@ -98,4 +100,51 @@ Array LogSoftmax(const Array& x, const OptionalAxes& axis) {
     return x_cast - LogSumExp(x_cast, axis.has_value() ? axis : OptionalAxes{1}, true);
 }
 
+Array Nansum(const Array& a, const OptionalAxes& axis, bool keepdims) {
+    Axes sorted_axis = internal::GetSortedAxesOrAll(axis, a.ndim());
+    Array a_masked = Where(IsNan(a), 0, a);
+    // Decide the output dtype for integral input dtype.
+    Dtype out_dtype{};
+    switch (GetKind(a_masked.dtype())) {
+        case DtypeKind::kBool:
+        case DtypeKind::kInt:  // fallthrough
+            out_dtype = Dtype::kInt64;
+            break;
+        case DtypeKind::kUInt:
+            out_dtype = Dtype::kInt64;  // TODO(niboshi): This should be kUInt64
+            break;
+        default:
+            out_dtype = a.dtype();
+    }
+
+    Array out = internal::EmptyReduced(a_masked.shape(), out_dtype, sorted_axis, keepdims, a_masked.device());
+    {
+        NoBackpropModeScope scope{};
+        a.device().backend().CallKernel<NansumKernel>(a_masked, sorted_axis, out);
+    }
+
+    BackwardBuilder bb{"nansum", a, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([sorted_axis, in_shape = a.shape(), keepdims](BackwardContext& bctx) {
+            const Array& gout = *bctx.output_grad();
+            CHAINERX_ASSERT(std::is_sorted(sorted_axis.begin(), sorted_axis.end()));
+
+            if (!(in_shape.ndim() == 0 || sorted_axis.empty() || keepdims)) {
+                Shape out_shape_broadcastable = gout.shape();
+                for (auto axis : sorted_axis) {
+                    out_shape_broadcastable.insert(out_shape_broadcastable.begin() + axis, 1);
+                }
+                Array input_grad = gout.Reshape(out_shape_broadcastable).BroadcastTo(in_shape);
+                bctx.input_grad() = Where(IsNan(input_grad), 0, input_grad);
+            } else {
+                Array input_grad = gout.BroadcastTo(in_shape);
+                bctx.input_grad() = Where(IsNan(input_grad), 0, input_grad);
```

Comment:
Could you do this like
```
gout = Where(IsNan(gout), 0, gout);
bctx.input_grad() = gout.BroadcastTo(in_shape);
```
This would result on better performance.

Reply:
Sorry for the delay in addressing the comment, I was facing certain installation issues.
Thanks for the suggestion @emcastillo . I have made the change. I am working on the tests and will add them soon.